File: dmn-test-cases/signavio/dmn/dmn2java/expected/singleton/java/dmn/example-credit-decision/MakeCreditDecisionRuleOutput.java

import java.util.*;

@javax.annotation.Generated(value = {"decisionTableRuleOutput.ftl", "makeCreditDecision"})
public class MakeCreditDecisionRuleOutput extends com.gs.dmn.runtime.RuleOutput {
    private String makeCreditDecision;

    public MakeCreditDecisionRuleOutput(boolean matched) {
        super(matched);
    }

    @com.fasterxml.jackson.annotation.JsonGetter("makeCreditDecision")
    public String getMakeCreditDecision() {
        return this.makeCreditDecision;
    }

    @com.fasterxml.jackson.annotation.JsonSetter("makeCreditDecision")
    public void setMakeCreditDecision(String makeCreditDecision) {
        this.makeCreditDecision = makeCreditDecision;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        MakeCreditDecisionRuleOutput other = (MakeCreditDecisionRuleOutput) o;
        if (this.getMakeCreditDecision() != null ? !this.getMakeCreditDecision().equals(other.getMakeCreditDecision()) : other.getMakeCreditDecision() != null) return false;

        return true;
    }

    @Override
    public int hashCode() {
        int result = 0;
        result = 31 * result + (this.getMakeCreditDecision() != null ? this.getMakeCreditDecision().hashCode() : 0);

        return result;
    }

    public String toString() {
        StringBuilder result_ = new StringBuilder("(matched=" + isMatched());
        result_.append(String.format(", makeCreditDecision='%s'", makeCreditDecision));
        result_.append(")");
        return result_.toString();
    }
}

File: dmn-test-cases/signavio/dmn/dmn2java/expected/proto/proto3/kotlin/dmn/example-credit-decision/MakeCreditDecisionRuleOutput.kt

@javax.annotation.Generated(value = ["decisionTableRuleOutput.ftl", "makeCreditDecision"])
class MakeCreditDecisionRuleOutput(matched: Boolean) : com.gs.dmn.runtime.RuleOutput(matched) {
    @com.fasterxml.jackson.annotation.JsonProperty("makeCreditDecision")
    var makeCreditDecision: String? = null

    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        if (javaClass != o?.javaClass) return false

        val other = o as MakeCreditDecisionRuleOutput
        if (if (this.makeCreditDecision != null) this.makeCreditDecision != other.makeCreditDecision else other.makeCreditDecision != null) return false

        return true
    }

    override fun hashCode(): Int {
        var result = 0
        result = 31 * result + (if (this.makeCreditDecision != null) this.makeCreditDecision.hashCode() else 0)

        return result
    }

    override fun toString(): String {
        val result_ = StringBuilder("(matched=" + isMatched())
        result_.append(String.format(", makeCreditDecision='%s'", makeCreditDecision))
        result_.append(")")
        return result_.toString()
    }
}
